File: generators/app/templates/middle-tier/src/main/java/javalin/controller/UserController.java
package <%= packageName %>.controller;

import io.javalin.http.Handler;
import java.util.List;
import java.util.Optional;

import <%= packageName %>.domain.User;
import <%= packageName %>.service.UserService;
import <%= packageName %>.service.UserServiceImpl;

/**
 * An example web service API controller.
 */
public class UserController {

    // Use static singleton component instances for services and repositories, or any dependency injection framework
    private static UserService userService = new UserServiceImpl();

    public static Handler users = ctx -> {
        // Add a fake delay so we can see state changes in the UI
        try {
            Thread.sleep(1000);
        }
        catch (InterruptedException e) {
        }
        ctx.json(userService.users());
    };

    public static Handler user = ctx -> {
        // Add a fake delay so we can see state changes in the UI
        try {
            Thread.sleep(1000);
        }
        catch (InterruptedException e) {
        }
        userService.user(ctx.pathParam("username")).ifPresentOrElse(ctx::json, () -> ctx.status(404));
    };

}

File: generators/app/templates/middle-tier/src/main/kotlin/javalin/controller/UserController.kt
package <%= packageName %>.controller

import io.javalin.http.Context

import <%= packageName %>.domain.User
import <%= packageName %>.service.UserServiceImpl

/**
 * An example web service API controller.
 */
object UserController {

    // Use static singleton component instances for services and repositories, or any dependency injection framework
    private val userService = UserServiceImpl()

    val users: (Context)->Unit = { ctx ->
        // Add a fake delay so we can see state changes in the UI
        try {
            Thread.sleep(1000)
        } catch (e: InterruptedException) {
        }

        ctx.json(userService.users())
    }

    val user: (Context)->Unit = { ctx ->
        // Add a fake delay so we can see state changes in the UI
        try {
            Thread.sleep(1000)
        } catch (e: InterruptedException) {
        }

        val user: User? = userService.user(ctx.pathParam("username"))
        if (user != null) ctx.json(user) else ctx.status(404)
    }

}
